client/src/components/ProductDetails.jsx

```diff
@@ -25,33 +26,104 @@ const useStyles = makeStyles({
     textDecoration: "underline"
   },
 
+  salePrice: {
+    color: "red"
+  },
+
+  lineThrough: {
+    textDecoration: "line-through",
+  },
+
   temp: {
     minHeight: "200px",
     border: "dotted 1px grey",
     marginBottom: "20px"
   }
 })
 
-const ProductDetails = () => {
+const getRatingInfo = (ratings) => {
+  let total = 0;
+  let count = 0;
+
+  for (let rating in ratings) {
+    let ratingInt = parseInt(rating)
+    let ratingVal = parseInt(ratings[rating])
+    total += ratingInt * ratingVal
+    count += ratingVal
+  }
+
+  console.log('avgRating:', (total / (count * 5)) * count);
+  console.log('totRatings:', count);
+
+  return {
+    avgProductRating: (total / (count * 5)) * count,
+    totalRatings: count
+  }
+};
+
+const ProductDetails = ({currentProduct, productStyles, styleIndex, changeStyle}) => {
   const classes = useStyles();
+  const [ratingsInfo, setRatingsInfo] = useState(null)
+
+  let productId = currentProduct.id;
+
+  useEffect(() => {
+    axios.get(`api/reviews/meta?product_id=${productId}`)
+    .then(({ data }) => {
+      setRatingsInfo(getRatingInfo(data.ratings))
+    }).catch((err) => {
+      console.log('Unable to retrieve ratings info', err);
+    })
+  }, [productId])
```

Comment:
Same comment here about the second argument in useEffect. I think it can just be an empty array

Reply:
iI only want the useEffect to run when the product ID changes, otherwise, updating state directly will will only change the details that are being displayed, but not make extraneous get requests to update that that will most likely stay the same.